File: xssfinder-executor-java/src/main/java/org/xssfinder/scanner/ThriftToReflectionLookup.java
package org.xssfinder.scanner;

import java.lang.reflect.Method;
import java.util.HashMap;
import java.util.Map;

public class ThriftToReflectionLookup {
    private final Map<String, Class<?>> pageClassLookup;
    private final Map<String, Method> methodLookup;

    public ThriftToReflectionLookup() {
        pageClassLookup = new HashMap<String, Class<?>>();
        methodLookup = new HashMap<String, Method>();
    }

    public Class<?> getPageClass(String pageId) {
        return pageClassLookup.get(pageId);
    }

    public Method getMethod(String methodId) {
        return methodLookup.get(methodId);
    }

    public void putPageClass(String pageId, Class<?> pageClass) {
        pageClassLookup.put(pageId, pageClass);
    }

    public void putMethod(String methodId, Method method) {
        methodLookup.put(methodId, method);
    }
}

File: xssfinder-php-executor/src/XssFinder/Scanner/ThriftToReflectionLookup.php
<?php

namespace XssFinder\Scanner;

use ReflectionClass;
use ReflectionMethod;

class ThriftToReflectionLookup
{
    private $_pageClassesById = array();
    private $_methodsById = array();

    /**
     * @param string $pageIdentifier
     * @return ReflectionClass|null
     */
    public function getPageClass($pageIdentifier)
    {
        return isset($this->_pageClassesById[$pageIdentifier]) ? $this->_pageClassesById[$pageIdentifier] : null;
    }

    /**
     * @param string $methodIdentifier
     * @return ReflectionMethod|null
     */
    public function getMethod($methodIdentifier)
    {
        return isset($this->_methodsById[$methodIdentifier]) ? $this->_methodsById[$methodIdentifier] : null;
    }

    /**
     * @param string $pageIdentifier
     * @param ReflectionClass $reflectionClass
     */
    public function putPageClass($pageIdentifier, ReflectionClass $reflectionClass)
    {
        $this->_pageClassesById[$pageIdentifier] = $reflectionClass;
    }

    /**
     * @param string $methodIdentifier
     * @param ReflectionMethod $method
     */
    public function putMethod($methodIdentifier, ReflectionMethod $method)
    {
        $this->_methodsById[$methodIdentifier] = $method;
    }
}